apps/bfd-shared-test-utils/src/main/java/gov/cms/bfd/AbstractLocalStackTest.java

```diff
@@ -0,0 +1,30 @@
+package gov.cms.bfd;
+
+import org.testcontainers.containers.localstack.LocalStackContainer;
+import org.testcontainers.utility.DockerImageName;
+
+/**
+ * Base class for tests that need to use a shared localstack container. Refer to <a
+ * href="https://java.testcontainers.org/test_framework_integration/manual_lifecycle_control/#singleton-containers">localstack
+ * docs</a> for explanation of the singleton container pattern.
+ */
+public abstract class AbstractLocalStackTest {
+  /** The localstack image and version to use for integration tests. */
+  public static final DockerImageName LocalStackImageName =
+      DockerImageName.parse("localstack/localstack:2.0.2");
```

Comment:
This might make the localstack image version harder to update; could/should this be read from the pom so it can be updated alongside other versioned items?

Reply:
Thats a good question.  I did some looking around in the past and did not find a way to get properties like this from the pom at compile time.  We do the same thing in `bfd-shared-test-utils/src/main/java/gov/cms/bfd/DatabaseTestUtils.java`.

An alternative might be to move all of the image names into a single class as constants so they can all be checked and updated in one place.

Reply:
Neither thing I found was particularly elegant, but there's the http://www.mojohaus.org/properties-maven-plugin/ that lets you write a maven plugin to be read in as a file which this could read from during setup, or there's setting a system var from maven using the surefire plugin: https://stackoverflow.com/questions/3231797/specify-system-property-to-maven-project

If we had the version saved as a maven property like the other versions, one of these two solutions could supply said version for consumption in the application. (There might be a better way that I didn't find, as well)

Reply:
While I applaud the goal I'm not a fan of using a system property for the image name/tag and storing the property value in the pom.  It works fine when running the tests from maven (admittedly our normal case) but if anyone wants to run the tests directly in their IDE without using maven they run into some issues:

When running the test directly in IDEA without maven you need to tell IDEA to set the property directly for the test, otherwise one of two things happen:
* If IDEA does not set the property and the java code has no default value for the property, the test will fail.
* If IDEA does not set the property and the java code has a default value that differs from the value in the pom the test will likely fail if there is some reason why we upgraded the image tag in the first place.

If the code does have a default value then we wind up having to maintain the image tag in two places (pom and java code) rather than one (java code).  Or even three if we include a developer's IDEA configuration.

So I'd prefer to just centralize the version strings in the java code and maybe make a note of it in the pom.